support/cas-server-support-inwebo-mfa/src/main/java/org/apereo/cas/support/inwebo/web/flow/actions/CheckAuthenticationAction.java

```diff
@@ -0,0 +1,72 @@
+package org.apereo.cas.support.inwebo.web.flow.actions;
+
+import org.apereo.cas.support.inwebo.authentication.InweboCredential;
+import org.apereo.cas.support.inwebo.service.InweboService;
+import org.apereo.cas.support.inwebo.service.response.Result;
+import org.apereo.cas.web.flow.resolver.CasWebflowEventResolver;
+import org.apereo.cas.web.support.WebUtils;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.context.MessageSource;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.webflow.action.AbstractAction;
+import org.springframework.webflow.execution.Event;
+import org.springframework.webflow.execution.RequestContext;
+
+/**
+ * A web action to check the push notification or the OTP generated on the browser.
+ *
+ * @author Jerome LELEU
+ * @since 6.4.0
+ */
+@RequiredArgsConstructor
+@Slf4j
+public class CheckAuthenticationAction extends AbstractAction implements WebflowConstants {
+
+    private final MessageSource messageSource;
```

Comment:
Can be obtained from the request-context. Shouldn't have to pass it in?

Reply:
I don't know how to do that. Can you elaborate? Thx

Reply:
@mmoayyed This is one of the latest point if you can provide me some guidance on this...

Reply:
Sorry I missed this.

To deal with the message bundles, you want to use `	RequestContext.getMessageContext` which will give you a message-context that you can fetch messages or add messages via keys. This is specially the recommended choice for SWF actions; you shouldn't have to directly deal with the message source itself.

Reply:
OK. Done. I have removed the injected `MessageSource` to use the `requestContext.getMessageContext()`.